src/lib/dom.js

```diff
@@ -237,11 +237,14 @@ export function extendUrl(url : string, params : { [key : string] : string } = {
 }
 
 export function redirect(win : CrossDomainWindowType = window, url : string) : ZalgoPromise<void> {
-    return new ZalgoPromise(resolve => {
+    return new ZalgoPromise((resolve, reject) => {
 
         info(`redirect`, { url });
 
         setTimeout(() => {
+            if (url.toLowerCase().replace(/[^a-z:]+/g, '').match(/^javascript:/)) {
+                return reject(new Error('Invalid redirect URL'));
+            }
```

Comment:
I spent an unreasonable amount of time trying to write a test for this, but left it out because:

- all of the tests have a custom `setupPopupBridge` implementation: https://github.com/paypal/paypal-checkout-components/blob/6cb970b6edbb5ac091e46444885ed5f1c99006ab/test/tests/button/popupBridge.js#L19
- that hardcode the `return_uri`: https://github.com/paypal/paypal-checkout-components/blob/6cb970b6edbb5ac091e46444885ed5f1c99006ab/test/tests/common.js#L623
- and depend on a `hashchange` event for assertions: https://github.com/paypal/paypal-checkout-components/blob/6cb970b6edbb5ac091e46444885ed5f1c99006ab/test/tests/button/popupBridge.js#L119-L121